src/components/data-grid/DataGrid.tsx

```diff
@@ -691,7 +706,7 @@ export const DataGrid = ({
             footerPageSize: FooterPageSize,
             noRowsTemplate: () => <NoRowsTemplate translations={translations} />,
             headerCheckbox: HeaderCheckbox,
-            headerColumnToggle: HeaderColumnToggle,
+            headerColumnToggle: () => <></>,
```

Comment:
I think this thing can be removed all together.

```suggestion
```

 but then it also has to be removed from line: `812` So its:

```
headerComponent={''}
```